src/web/rustdoc.rs

```diff
@@ -209,141 +209,158 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
-        return redirect(&name, &version, &req_path[4..]);
+    // Create the path to access the file from
+    let mut path = req_path.join("/");
+    if path.ends_with('/') {
+        req_path.pop(); // get rid of empty string
+        path.push_str("index.html");
+        req_path.push("index.html");
     }
 
-    let mut path = {
-        let mut path = req_path.join("/");
-        if path.ends_with('/') {
-            req_path.pop(); // get rid of empty string
-            path.push_str("index.html");
-            req_path.push("index.html");
-        }
-        path
-    };
+    // Attempt to load the file from the database
+    let file = if let Some(file) = File::from_path(&conn, &path) {
+        file
+    } else {
+        // If it fails, we try again with /index.html at the end
+        path.push_str("/index.html");
+        req_path.push("index.html");
 
-    let file = match File::from_path(&conn, &path) {
-        Some(f) => f,
-        None => {
-            // If it fails, we try again with /index.html at the end
-            path.push_str("/index.html");
-            req_path.push("index.html");
-            match File::from_path(&conn, &path) {
-                Some(f) => f,
-                None => return Err(IronError::new(Nope::ResourceNotFound, status::NotFound)),
-            }
-        }
+        File::from_path(&conn, &path)
+            .ok_or_else(|| IronError::new(Nope::ResourceNotFound, status::NotFound))?
     };
 
-    let req_path = req_path;
-
-    // serve file directly if it's not html
+    // Serve non-html files immediately
     if !path.ends_with(".html") {
         return Ok(file.serve());
     }
 
-    let mut content = RustdocPage::default();
-
     let file_content = ctry!(String::from_utf8(file.0.content));
-
+    // Extract the head and body of the rustdoc file so that we can insert it into our own html
     let (head, body, mut body_class) = ctry!(utils::extract_head_and_body(&file_content));
-    content.head = head;
-    content.body = body;
 
+    // Add the `rustdoc` classes to the html body
     if body_class.is_empty() {
         body_class = "rustdoc container-rustdoc".to_string();
     } else {
-        // rustdoc adds its own "rustdoc" class to the body
         body_class.push_str(" container-rustdoc");
     }
-    content.body_class = body_class;
-
-    content.full = file_content;
 
+    // Get the latest version of the crate
     let latest_version = crate_details.latest_version().to_owned();
     let is_latest_version = latest_version == version;
 
+    // If the requested version is not the latest, then find the path of the latest version for easy redirection
     let path_in_latest = if !is_latest_version {
+        // Replace the version of the old path with the latest version
         let mut latest_path = req_path.clone();
         latest_path[2] = &latest_version;
+
+        // Retrieve the path in the database of the latest crate version
```

Comment:
I think this is a _little_ overboard with the comments :P. If you want to improve the comments on `path_for_version` though, that would be much appreciated. The point of that function is to search for the current page if it doesn't exist in the latest version.
```suggestion

```